### PR TITLE
Add ability to set oAuth config for CHE deployed to Openshift

### PR DESCRIPTION
**Pull request by riuvshin**
Add ability to set oAuth config for CHE deployed to Openshift, also changed ENV var `CHE_MULTI_USER` to `CHE_MULTIUSER` as it is in CHE

needed for: https://github.com/eclipse/che/issues/7211

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
I would say that CHE_MULTI_USER is more natural to me than CHE_MULTIUSER

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 1]
```diff
@@ -397,7 +399,13 @@ MULTI_USER_REPLACEMENT_STRING="          - name: \"CHE_WORKSPACE_LOGS\"
           - name: \"CHE_KEYCLOAK_CLIENT__ID\"
             value: \"${CHE_KEYCLOAK_CLIENT__ID}\"
           - name: \"CHE_HOST\"
-            value: \"${CHE_HOST}\""
+            value: \"${CHE_HOST}\"
+          - name: \"CHE_MULTIUSER\"
+            value: \"${CHE_MULTIUSER}\"
+          - name: \"CHE_OAUTH_GITHUB_CLIENTID\"
+            value: \"${CHE_OAUTH_GITHUB_CLIENTID}\"
```
Comment (garagatyi) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:406]: OAUTH credentials don't look related to multi-user case. Have you considered using it in another context?

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 2]
```diff
@@ -397,7 +399,13 @@ MULTI_USER_REPLACEMENT_STRING="          - name: \"CHE_WORKSPACE_LOGS\"
           - name: \"CHE_KEYCLOAK_CLIENT__ID\"
             value: \"${CHE_KEYCLOAK_CLIENT__ID}\"
           - name: \"CHE_HOST\"
-            value: \"${CHE_HOST}\""
+            value: \"${CHE_HOST}\"
+          - name: \"CHE_MULTIUSER\"
+            value: \"${CHE_MULTIUSER}\"
```
Comment (garagatyi) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:404]: Can you elaborate on how this value is used?

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 3]
```diff
@@ -303,7 +305,7 @@ fi
 
 COMMAND_DIR=$(dirname "$0")
 
-if [ "${CHE_MULTI_USER}" == "true" ]; then
+if [ "${CHE_MULTIUSER}" == "true" ]; then
```
Comment (garagatyi) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:308]: +1 to sync this variable name and variable name we are already using in Che server/CLI

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 2 above)
Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:404]: that would be needed for https://github.com/eclipse/che/pull/7188
depending on CHE_MULTIUSER true || false tomcat will be launched in appropriate mode.

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 1 above)
Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:406]: 😱 😱 😱 +100

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 2 above)
Comment (garagatyi) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:404]: ok, thanks for the explanation

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 1 above)
Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:406]: I've renamed VAR from `MULTI_USER_REPLACEMENT_STRING` to `CHE_SERVER_CONFIGURATION`
because now it is not in multi / single user context and used for each usecase.
And it was not replacement string already :) it was just yaml injection

Comment (garagatyi) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:406]: ok

**Review by garagatyi — APPROVED** [review]

**Review by ibuziuk — APPROVED** [review]

**Review by l0rd — APPROVED** [review]
I think @kameshsampath will be happy when this one will be merged

**Review by davidfestal — APPROVED** [review]

**Comment by riuvshin** [comment]
@l0rd please note that is che6 only bracnh. 
I didn't planned to do that in master, should I ?

**Comment by kameshsampath** [comment]
yeah thanks @l0rd  :)

**Comment by l0rd** [comment]
@riuvshin that would be cool so that @kameshsampath would be able to use it on minishift right away.
